packages/inventory-next/src/services/inventory.ts

```diff
@@ -0,0 +1,1053 @@
+import { DAL } from "@medusajs/types"
+import { InjectTransactionManager } from "@medusajs/utils"
+import { InjectManager } from "@medusajs/utils"
+import InventoryLevelService from "./inventory-level"
+import { partitionArray } from "@medusajs/utils"
+import { InventoryEvents } from "@medusajs/utils"
+import { CommonEvents } from "@medusajs/utils"
+import { isDefined } from "@medusajs/utils"
+
+type InjectedDependencies = {
+  baseRepository: DAL.RepositoryService
+  inventoryItemService: ModulesSdkTypes.InternalModuleService<any>
+  inventoryLevelService: InventoryLevelService<any>
+  reservationItemService: ModulesSdkTypes.InternalModuleService<any>
+}
+
+const generateMethodForModels = [
+  { model: InventoryItem, singular: "InventoryItem", plural: "InventoryItems" },
+  {
+    model: InventoryLevel,
+    singular: "InventoryLevel",
+    plural: "InventoryLevels",
+  },
+  {
+    model: ReservationItem,
+    singular: "ReservationItem",
+    plural: "ReservationItems",
+  },
+]
+
+export default class InventoryModuleService<
+    TInventoryItem extends InventoryItem = InventoryItem,
+    TInventoryLevel extends InventoryLevel = InventoryLevel,
+    TReservationItem extends ReservationItem = ReservationItem
+  >
+  extends ModulesSdkUtils.abstractModuleServiceFactory<
+    InjectedDependencies,
+    InventoryNext.InventoryItemDTO,
+    {
+      InventoryItem: {
+        dto: InventoryNext.InventoryItemDTO
+        singular: "InventoryItem"
+        plural: "InventoryItems"
+      }
+      InventoryLevel: {
+        dto: InventoryNext.InventoryLevelDTO
+        singular: "InventoryLevel"
+        plural: "InventoryLevels"
+      }
+      ReservationItem: {
+        dto: InventoryNext.ReservationItemDTO
+        singular: "ReservationItem"
+        plural: "ReservationItems"
+      }
+    }
+  >(InventoryItem, generateMethodForModels, entityNameToLinkableKeysMap)
+  implements IInventoryServiceNext
+{
+  protected baseRepository_: DAL.RepositoryService
+
+  protected readonly inventoryItemService_: ModulesSdkTypes.InternalModuleService<TInventoryItem>
+  protected readonly reservationItemService_: ModulesSdkTypes.InternalModuleService<TReservationItem>
+  protected readonly inventoryLevelService_: InventoryLevelService<TInventoryLevel>
+
+  constructor(
+    {
+      baseRepository,
+      inventoryItemService,
+      inventoryLevelService,
+      reservationItemService,
+    }: InjectedDependencies,
+    protected readonly moduleDeclaration?: InternalModuleDeclaration
+  ) {
+    // @ts-ignore
+    // eslint-disable-next-line prefer-rest-params
+    super(...arguments)
+
+    this.baseRepository_ = baseRepository
+    this.inventoryItemService_ = inventoryItemService
+    this.inventoryLevelService_ = inventoryLevelService
+    this.reservationItemService_ = reservationItemService
+  }
+
+  __joinerConfig(): ModuleJoinerConfig {
+    return joinerConfig
+  }
+
+  @InjectTransactionManager("baseRepository_")
```

Comment:
q: do we need that on private method?

Reply:
probably not as this should always have a transaction if it's called 😅 I'll remove it